app/models/academic_year.rb

```diff
@@ -13,4 +13,28 @@ class AcademicYear < ApplicationRecord
   validates :name, presence: true
   validates :starts_on, presence: true
   validates :ends_on, presence: true, comparison: { greater_than_or_equal_to: :starts_on }
+
+  START_DATE = {
+    day: 1,
+    month: 9, # September
+  }.freeze
+
+  END_DATE = {
+    day: 31,
+    month: 8, # August
+  }.freeze
+
+  def self.for_date(date)
```

Comment:
Naming this something along the lines of `create_for_date` instead of `for_date` should help reveal the method's intentions.

Personally, I'd expect `for_date` query to find a record for a given date, rather than create a record for the date.

Reply:
@Nitemaeric the naming and design of this interface is intentional. There are [more details in the Trello card](https://trello.com/c/qLLh08g6/635-refactor-the-academic-year-model-to-consolidate-hardcoded-magic-dates), which I believe you already read and agreed to.

The TL;DR is that `.for_date` should be the primary interface for getting the Academic Year for a given date. This directly replaces the use of `find_or_create_by` that was present throughout the code.

The fact that it 'creates' or 'finds' a record is an internal implementation detail, but it's not consequential or relevant to consumers. So this implementation intentionally abstracts those details away.

As a public interface, consumers should be able to call `.for_date` and receive an Academic Year object which exists in the database – but _how_ the record was found is not relevant here. So in this case, I think `.for_date` is the appropriate name for the method.